pkg/weather-client/client.go

```diff
@@ -83,10 +28,37 @@ func New() WeatherClient {
 	return newNoaaClient()
 }
 
-func (nc *NoaaClient) GetForecast(xCoordinate float32, yCoordinate float32) (*ForecastResponse, error) {
-	// BOU weather office(boulder), 52X, 75Y this is boulder
-	url := fmt.Sprintf("%s/gridpoints/BOU/%f,%f/forecast/hourly",
-		nc.APIEndpoint, xCoordinate, yCoordinate)
+func (nc *NoaaClient) getNoaaPointInfo(lat float64, lng float64) (xCoord float64, yCoord float64, stationCode string, err error) {
```

Comment:
Seems code would be cleaner if you returned `*PointResponse` here?

```go
func (nc *NoaaClient) getNoaaPointInfo(lat float64, lng float64) (*PointResponse, error) {
```

This would also simplify your error returns to `return nil, err`.

Most importantly, you're using **named result parameters** here, namely things like `xCoord float64` instead of just `float64`. This is a fairly uncommon practice. Named result parameters actually **initialize** those named variables at the start of the function/method. Meaning you have a variable `xCoord`, available right at the start. The idea is that you can now assign `xCoord = float64(pointData.Properties.GridX)` at some point. Then when the function exits or you call `return`, the value of `xCoord` is returned as the first return parameter. So concretely, this would change to:

```go
xCoord = float64(pointData.Properties.GridX)
yCoord = float64(pointData.Properties.GridY)
stationCode = pointData.Properties.Cwa
err = nil

// more code if there were any

return
```

In short, generally this isn't something you'll do and/or you'll know when you need it. If for any reason you want to help the client (caller) identify what the return parameters represent its best to:

* a consider if you should be returning a more clear structured type (which is my suggestion with `*PointResponse`.

  -- or --

* use documentation (comments) to describe the return parameters.

Reply:
Gotcha, my intention was to make the function return a simple version of what we need as opposed to a vomit of information from the points endpoint.

I can just adjust the type so it doesn't unmarshall a bunch of stuff when we convert the json.

Reply:
quick question, 

you're suggesting `func (nc *NoaaClient) getNoaaPointInfo(lat float64, lng float64) (*PointResponse, error) {`

why not make it return a non-pointer PointResponse? We don't need to actually pass the original right? Is it typically best practice to default to passing a pointer and only pass the copied object under specific circumstances?